unitary/examples/quantum_checkers/checkers.py

```diff
@@ -0,0 +1,311 @@
+        if self.board_matrix[to_row][to_col].occupant.king == False and (
+            to_row == 0 or to_row == self.num_rows - 1
+        ):
+            self.board_matrix[to_row][to_col].occupant.king = True
+        return
+
+    def remove_piece(self, row, col):
+        self.board_matrix[row][col].occupant = None
+        return
+
+    def print_board(self):
+        output = "\n"
+        output += "   |"
+        for i in range(self.num_cols):
+            output += f" {i} |"
+        output += "\n" + "---|" * (self.num_cols + 1)
+        output += "\n"
+        for i in range(self.num_rows):
+            output += f" {i} |"
+            for j in range(self.num_cols):
+                # print(f"{self.board_matrix[i][j].occupant.color}")
+                if self.board_matrix[i][j].occupant != None:
+                    # print(self.board_matrix[i][j].occupant.color)
+                    if self.board_matrix[i][j].occupant.color == Colors.WHITE:
+                        if self.board_matrix[i][j].occupant.king == True:
+                            output += " W "
+                        else:
+                            output += " w "
+                    else:
+                        if self.board_matrix[i][j].occupant.king:
+                            output += " B "
+                        else:
+                            output += " b "
+                    # print(type(self.board_matrix[i][j].occupant))
+                    # output += f" {self.board_matrix[i][j].occupant.color} "
+                else:
+                    output += "   "
+                output += "|"
+            output += "\n" + "---|" * (self.num_cols + 1)
+            output += "\n"
+        return output
+
+    def calculate_all_possible_moves(self, color: Colors):
+        legal_moves = []
+        legal_take_moves = []
+        # Loop over all squares, if there is a piece there check what moves are possible.
+        # Moves will be
+        for i in range(self.num_rows):
+            for j in range(self.num_cols):
+                if (
+                    self.board_matrix[i][j].occupant != None
+                    and self.board_matrix[i][j].occupant.color == color
+                ):
+                    # temp1, temp2 = self.calculate_legal_move(i, j, color)
+                    # legal_moves.extend(temp1)
+                    # legal_take_moves.extend(temp2)
+                    temp_legal_moves, temp_take_moves = self.possible_moves(i, j)
+                    legal_moves.extend(temp_legal_moves)
+                    legal_take_moves.extend(temp_take_moves)
+        if len(legal_take_moves) != 0 and forced_take:
+            return legal_take_moves
+        return legal_moves
+
+    def on_board(self, row, col):
+        """
+        Checks if given location is on the board on not.
+        Returns true if [row][col] is on the board
+        """
+        if row < 0 or row > self.num_rows - 1 or col < 0 or col > self.num_cols - 1:
```

Comment:
nit: can simplify this:

`return row < self.num_rows and row>=0 and col < self.num_cols and col>=0`